File: Java/Szyfrowanie ROT13.java
import java.util.Scanner;

public class Main 
{
    public static char rot13(char c) 
    {
        if (c >= 'a' && c <= 'z')
            return (char) ((c - 'a' + 13) % 26 + 'a');
        else if (c >= 'A' && c <= 'Z')
            return (char) ((c - 'A' + 13) % 26 + 'A');
        else if (c >= '0' && c <= '9')
            return (char) ((c - '0' + 5) % 10 + '0');
        else
            return c;
    }

    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        while (input.hasNextLine()) 
        {
            String line = input.nextLine();
            for (char c : line.toCharArray())
                System.out.print(rot13(c));
            System.out.println();
        }
        input.close();
    }
}

File: C++/Szyfrowanie ROT13.cpp
#include <string>
#include <iostream>

const char rot13(char& c)
{
    if (c >= 'a' && c <= 'z')
        return (c - 'a' + 13) % 26 + 'a';
    else if (c >= 'A' && c <= 'Z')
        return (c - 'A' + 13) % 26 + 'A';
    else if (c >= '0' && c <= '9')
        return (c - '0' + 5) % 10 + '0';
    else
        return c;
}

int main()
{
    std::string line;
    while (std::cin >> line)
    {
        for (char& c : line)
            std::cout << rot13(c);
        std::cout << '\n';
    }
    return 0;
}
